pkg/services/ngalert/README.md

```diff
@@ -0,0 +1,87 @@
+# Next generation alerting (ngalert) in Grafana 8
+
+Ngalert (Next generation alert) is the next generation of alerting in Grafana 8.
+
+## Overview
+
+The ngalert package can be found in `pkg/services/ngalert` and has the following sub-packages:
+
+    - api
+    - eval
+    - logging
+    - metrics
+    - models
+    - notifier
+    - schedule
+    - sender
+    - state
+        - The state package does XYZ
+    - store
+    - tests
+
+## Scheduling and evaluation of alert rules
+
+The scheduling of alert rules happens in the `schedule` package. This package is responsible for managing the evaluation
+of alert rules including checking for new alert rules and stopping the evaluation of deleted alert rules.
+
+The scheduler runs at a fixed interval, called its heartbeat, in which it does a number of tasks:
+
+1. Fetch the alert rules for all (excluding disabled) organizations
+2. Start a goroutine (if this is a new alert rule or the scheduler has just started) to evaluate the alert rule
+3. Send a `*evalContext` event to the goroutine for each alert rule if its interval has elapsed
+4. Stops the goroutines for all alert rules that have been deleted since the last heartbeat
+
+The function that evaluates each alert rule is called `ruleRoutine`. It waits for an `*evalContext` event (sent each
+interval seconds elapsed and is configurable per alert rule) and then evaluates the alert rule. To ensure that the
+scheduler is evaluating the latest version of the alert rule it compares its local version of the alert rule with that
+in the `*evalContext` event, and, if the version numbers mismatch gets the latest version of the alert rule from the
```

Comment:
I would delimitate the apposition with a comma and use a period and start a new sentence, e.g. "...it compares its local version of the alert rule with that in the `*evalContext` event. If the version numbers mismatch, it gets the latest version..."